src/pudl/metadata/classes.py

```diff
@@ -1001,6 +1058,69 @@ class ResourceHarvest(PudlMeta):
     """Fraction of invalid fields above which result is considerd invalid."""
 
 
+class PudlResourceDescriptor(PudlMeta):
```

Comment:
Added this class so that we have a proper description/enforcement of the shape of `RESOURCE_METADATA`s. This should make it easier for someone to understand how to add a new resource.

Ran into a couple things that are a little funky, documented as TODOs.